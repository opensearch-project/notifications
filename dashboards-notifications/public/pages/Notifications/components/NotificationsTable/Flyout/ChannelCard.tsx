/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import {
  EuiCard,
  EuiDescriptionList,
  EuiFlexGroup,
  EuiFlexItem,
  EuiHealth,
  EuiLink,
  EuiText,
} from '@elastic/eui';
import _ from 'lodash';
import React from 'react';
import { ChannelStatus } from '../../../../../../models/interfaces';
import { isStatusCodeSuccess } from '../../../../../services/utils/helper';
import { CHANNEL_TYPE, ROUTES } from '../../../../../utils/constants';

interface ChannelCardProps {
  channel: ChannelStatus;
  onClose: () => void;
}

export function ChannelCard(props: ChannelCardProps) {
  const renderList = (
    title: NonNullable<React.ReactNode>,
    description: NonNullable<React.ReactNode>
  ) => {
    return (
      <EuiDescriptionList
        titleProps={{
          style: {
            color: '#69707D',
            fontSize: '0.6875rem',
            fontWeight: 'normal',
          },
        }}
        listItems={[{ title, description }]}
      />
    );
  };

  const renderStatus = (status: string) => {
    const success = isStatusCodeSuccess(status);
    const color = success ? 'success' : 'danger';
    const label = success ? 'Sent' : 'Error';
    return (
      <EuiHealth color={color}>
        <EuiText size="s">{label}</EuiText>
      </EuiHealth>
    );
  };

  return (
    <>
      <EuiCard
        textAlign="left"
        title={
          <EuiLink
            onClick={() => {
              location.assign(
                `#${ROUTES.CHANNEL_DETAILS}/${props.channel.config_id}`
              );
              props.onClose();
            }}
          >
            {props.channel.config_name}
          </EuiLink>
        }
        titleSize="xs"
        description=""
      >
        <EuiFlexGroup>
          <EuiFlexItem>
            {renderList(
              'Channel type',
              _.get(CHANNEL_TYPE, props.channel.config_type, '-')
            )}
          </EuiFlexItem>
          <EuiFlexItem>
            {renderList(
              'Sent status',
              renderStatus(props.channel.delivery_status.status_code)
            )}
          </EuiFlexItem>
        </EuiFlexGroup>
        {!isStatusCodeSuccess(props.channel.delivery_status.status_code) &&
          renderList(
            'Error details',
            props.channel.delivery_status.status_text
          )}
      </EuiCard>
    </>
  );
}
