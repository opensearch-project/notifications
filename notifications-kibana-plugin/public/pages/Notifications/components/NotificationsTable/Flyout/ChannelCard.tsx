/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import {
  EuiCard,
  EuiDescriptionList,
  EuiFlexGroup,
  EuiFlexItem,
  EuiHealth,
  EuiLink,
  EuiText,
} from '@elastic/eui';
import React from 'react';
import { ROUTES } from '../../../../../utils/constants';

interface ChannelCardProps {}

export function ChannelCard(props: ChannelCardProps) {
  const renderList = (
    title: NonNullable<React.ReactNode>,
    description: NonNullable<React.ReactNode>
  ) => {
    return (
      <EuiDescriptionList
        titleProps={{
          style: {
            color: '#69707D',
            fontSize: '0.6875rem',
            fontWeight: 'normal',
          },
        }}
        listItems={[{ title, description }]}
      />
    );
  };

  const renderStatus = (status: 'Success' | 'Error') => {
    const color = status == 'Success' ? 'success' : 'danger';
    const label = status == 'Success' ? 'Sent' : 'Errors';
    return (
      <EuiHealth color={color}>
        <EuiText size="s">{label}</EuiText>
      </EuiHealth>
    );
  };

  return (
    <>
      <EuiCard
        textAlign="left"
        title={
          <EuiLink href={`#${ROUTES.CHANNEL_DETAILS}/${'id'}`}>
            {'Managers'}
          </EuiLink>
        }
        titleSize="xs"
        description=""
      >
        <EuiFlexGroup>
          <EuiFlexItem>{renderList('Channel type', 'Email')}</EuiFlexItem>
          <EuiFlexItem>
            {renderList('Sent status', renderStatus('Success'))}
          </EuiFlexItem>
        </EuiFlexGroup>
        {renderList('Error details', 'unavailable')}
      </EuiCard>
    </>
  );
}
