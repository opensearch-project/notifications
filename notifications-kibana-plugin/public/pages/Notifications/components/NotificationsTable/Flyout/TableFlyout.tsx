/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import {
  EuiFlyout,
  EuiFlyoutBody,
  EuiFlyoutHeader,
  EuiTitle,
} from '@elastic/eui';
import React from 'react';

interface TableFlyoutProps {
  flyoutOpen: boolean;
  setFlyoutOpen: (flyoutOpen: boolean) => void;
}

export function TableFlyout(props: TableFlyoutProps) {
  return (
    <>
      {props.flyoutOpen ? (
        <EuiFlyout size="s" onClose={() => props.setFlyoutOpen(false)}>
          <EuiFlyoutHeader hasBorder>
            <EuiTitle>
              <h2>Notification details</h2>
            </EuiTitle>
          </EuiFlyoutHeader>
          <EuiFlyoutBody>hello</EuiFlyoutBody>
        </EuiFlyout>
      ) : null}
    </>
  );
}
