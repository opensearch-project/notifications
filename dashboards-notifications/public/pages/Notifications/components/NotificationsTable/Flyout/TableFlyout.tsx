/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import {
  EuiButtonEmpty,
  EuiDescriptionList,
  EuiFlexGroup,
  EuiFlexItem,
  EuiFlyout,
  EuiFlyoutBody,
  EuiFlyoutFooter,
  EuiFlyoutHeader,
  EuiLink,
  EuiSpacer,
  EuiTitle,
} from '@elastic/eui';
import _ from 'lodash';
import React from 'react';
import { NotificationItem } from '../../../../../../models/interfaces';
import { ModalRootProps } from '../../../../../components/Modal/ModalRoot';
import { NOTIFICATION_SOURCE } from '../../../../../utils/constants';
import { renderTime } from '../../../../../utils/helpers';
import { ChannelCard } from './ChannelCard';

interface TableFlyoutProps extends ModalRootProps {
  notificationItem: NotificationItem;
  onClose: () => void;
}

export function TableFlyout(props: TableFlyoutProps) {
  return (
    <>
      <EuiFlyout size="s" onClose={props.onClose}>
        <EuiFlyoutHeader hasBorder>
          <EuiTitle>
            <h2>Notification details</h2>
          </EuiTitle>
        </EuiFlyoutHeader>
        <EuiFlyoutBody>
          <EuiDescriptionList
            listItems={[
              {
                title: 'Notification',
                description: props.notificationItem.title,
              },
            ]}
          />
          <EuiSpacer />
          <EuiFlexGroup>
            <EuiFlexItem>
              <EuiDescriptionList
                listItems={[
                  {
                    title: 'Source',
                    // TODO source plugin link
                    description: (
                      <EuiLink href="#" target="_blank" external>
                        new monitor
                      </EuiLink>
                    ),
                  },
                ]}
              />
            </EuiFlexItem>
            <EuiFlexItem>
              <EuiDescriptionList
                listItems={[
                  {
                    title: 'Source type',
                    description: _.get(
                      NOTIFICATION_SOURCE,
                      props.notificationItem.source,
                      '-'
                    ),
                  },
                ]}
              />
            </EuiFlexItem>
          </EuiFlexGroup>
          <EuiSpacer />
          <EuiDescriptionList
            listItems={[
              {
                title: 'Time sent',
                description: renderTime(props.notificationItem.lastUpdatedTime),
              },
            ]}
          />
          <EuiSpacer />
          <EuiTitle size="xs">
            <h4>Channels sent</h4>
          </EuiTitle>
          {props.notificationItem.statusList.map((channelStatus) => (
            <div key={`channel-card-${channelStatus.configId}`}>
              <EuiSpacer size="s" />
              <ChannelCard channel={channelStatus} onClose={props.onClose} />
            </div>
          ))}
        </EuiFlyoutBody>
        <EuiFlyoutFooter>
          <EuiButtonEmpty iconType="cross" onClick={props.onClose} flush="left">
            Close
          </EuiButtonEmpty>
        </EuiFlyoutFooter>
      </EuiFlyout>
    </>
  );
}
