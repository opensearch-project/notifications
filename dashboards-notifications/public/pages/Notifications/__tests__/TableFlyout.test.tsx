/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import { render } from '@testing-library/react';
import React from 'react';
import { notificationServiceMock } from '../../../../test/mocks/serviceMock';
import { ChannelCard } from '../components/NotificationsTable/Flyout/ChannelCard';
import { TableFlyout } from '../components/NotificationsTable/Flyout/TableFlyout';

describe('<TableFlyout /> spec', () => {
  const item = {
    id: '1',
    title: 'Alert notification on high error rate',
    referenceId: 'alert_id_1',
    source: 'Alerting',
    severity: 'High',
    lastUpdatedTime: 1612229000,
    tags: ['optional string list'],
    status: 'Error',
    statusList: [
      {
        configId: '1',
        configName: 'dev_email_channel',
        configType: 'Email',
        emailRecipientStatus: [
          {
            recipient: 'dd@amazon.com',
            deliveryStatus: { statusCode: '500', statusText: 'Some error' },
          },
          {
            recipient: 'cc@amazon.com',
            deliveryStatus: { statusCode: '404', statusText: 'invalid' },
          },
        ],
        deliveryStatus: { statusCode: '500', statusText: 'Error' },
      },
      {
        configId: '2',
        configName: 'manage_slack_channel',
        configType: 'Slack',
        deliveryStatus: { statusCode: '200', statusText: 'Success' },
      },
    ],
  };

  it('renders the component', () => {
    const utils = render(
      <TableFlyout
        notificationItem={item}
        onClose={() => {}}
        services={notificationServiceMock}
      />
    );
    expect(utils.container.firstChild).toMatchSnapshot();
  });

  it('clicks card link', () => {
    const channel = item.statusList[0];
    const onClose = jest.fn();
    const utils = render(
      <ChannelCard channel={channel} onClose={onClose} />
    );
    utils.getByText(channel.configName).click();
    expect(onClose).toBeCalled();
  });
});
