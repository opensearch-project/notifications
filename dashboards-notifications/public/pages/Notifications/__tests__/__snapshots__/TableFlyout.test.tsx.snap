// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableFlyout /> spec renders the component 1`] = `
<div>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="1"
  />
  <div
    data-focus-lock-disabled="false"
  >
    <div
      class="euiFlyout euiFlyout--small"
      role="dialog"
      tabindex="0"
    >
      <button
        aria-label="Close this dialog"
        class="euiButtonIcon euiButtonIcon--text euiFlyout__closeButton"
        data-test-subj="euiFlyoutCloseButton"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonIcon__icon"
          focusable="false"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        />
      </button>
      <div
        class="euiFlyoutHeader euiFlyoutHeader--hasBorder"
      >
        <h2
          class="euiTitle euiTitle--medium"
        >
          Notification details
        </h2>
      </div>
      <div
        class="euiFlyoutBody"
      >
        <div
          class="euiFlyoutBody__overflow"
        >
          <div
            class="euiFlyoutBody__overflowContent"
          >
            <dl
              class="euiDescriptionList euiDescriptionList--row"
            >
              <dt
                class="euiDescriptionList__title"
              >
                Notification
              </dt>
              <dd
                class="euiDescriptionList__description"
              >
                test notification
              </dd>
            </dl>
            <div
              class="euiSpacer euiSpacer--l"
            />
            <div
              class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
            >
              <div
                class="euiFlexItem"
              >
                <dl
                  class="euiDescriptionList euiDescriptionList--row"
                >
                  <dt
                    class="euiDescriptionList__title"
                  >
                    Source
                  </dt>
                  <dd
                    class="euiDescriptionList__description"
                  >
                    <a
                      class="euiLink euiLink--primary"
                      href="reports-dashboards#/report_details/reference_id"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Report
                      <svg
                        aria-hidden="true"
                        aria-label="External link"
                        class="euiIcon euiIcon--small euiIcon-isLoading euiLink__externalIcon"
                        focusable="false"
                        height="16"
                        role="img"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      />
                    </a>
                  </dd>
                </dl>
              </div>
              <div
                class="euiFlexItem"
              >
                <dl
                  class="euiDescriptionList euiDescriptionList--row"
                >
                  <dt
                    class="euiDescriptionList__title"
                  >
                    Source type
                  </dt>
                  <dd
                    class="euiDescriptionList__description"
                  >
                    Reporting
                  </dd>
                </dl>
              </div>
            </div>
            <div
              class="euiSpacer euiSpacer--l"
            />
            <dl
              class="euiDescriptionList euiDescriptionList--row"
            >
              <dt
                class="euiDescriptionList__title"
              >
                Time sent
              </dt>
              <dd
                class="euiDescriptionList__description"
              >
                07/06/03 11:42 am PDT
              </dd>
            </dl>
            <div
              class="euiSpacer euiSpacer--l"
            />
            <h4
              class="euiTitle euiTitle--xsmall"
            >
              Channels sent
            </h4>
            <div>
              <div
                class="euiSpacer euiSpacer--s"
              />
              <div
                class="euiCard euiCard--paddingMedium euiCard--leftAligned euiCard--hasChildren"
              >
                <div
                  class="euiCard__content"
                >
                  <span
                    class="euiTitle euiTitle--xsmall euiCard__title"
                    id="random_html_idTitle"
                  >
                    <button
                      class="euiLink euiLink--primary"
                      type="button"
                    >
                      slack test
                    </button>
                  </span>
                  <div
                    class="euiText euiText--small euiCard__description"
                    id="random_html_idDescription"
                  >
                    <p />
                  </div>
                  <div
                    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
                  >
                    <div
                      class="euiFlexItem"
                    >
                      <dl
                        class="euiDescriptionList euiDescriptionList--row"
                      >
                        <dt
                          class="euiDescriptionList__title"
                          style="color: rgb(105, 112, 125); font-size: 0.6875rem; font-weight: normal;"
                        >
                          Channel type
                        </dt>
                        <dd
                          class="euiDescriptionList__description"
                        >
                          slack
                        </dd>
                      </dl>
                    </div>
                    <div
                      class="euiFlexItem"
                    >
                      <dl
                        class="euiDescriptionList euiDescriptionList--row"
                      >
                        <dt
                          class="euiDescriptionList__title"
                          style="color: rgb(105, 112, 125); font-size: 0.6875rem; font-weight: normal;"
                        >
                          Sent status
                        </dt>
                        <dd
                          class="euiDescriptionList__description"
                        >
                          <div
                            class="euiHealth"
                          >
                            <div
                              class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
                            >
                              <div
                                class="euiFlexItem euiFlexItem--flexGrowZero"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="euiIcon euiIcon--medium euiIcon--success euiIcon-isLoading"
                                  focusable="false"
                                  height="16"
                                  role="img"
                                  viewBox="0 0 16 16"
                                  width="16"
                                  xmlns="http://www.w3.org/2000/svg"
                                />
                              </div>
                              <div
                                class="euiFlexItem euiFlexItem--flexGrowZero"
                              >
                                <div
                                  class="euiText euiText--small"
                                >
                                  Sent
                                </div>
                              </div>
                            </div>
                          </div>
                        </dd>
                      </dl>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <div
                class="euiSpacer euiSpacer--s"
              />
              <div
                class="euiCard euiCard--paddingMedium euiCard--leftAligned euiCard--hasChildren"
              >
                <div
                  class="euiCard__content"
                >
                  <span
                    class="euiTitle euiTitle--xsmall euiCard__title"
                    id="random_html_idTitle"
                  >
                    <button
                      class="euiLink euiLink--primary"
                      type="button"
                    >
                      emailtest
                    </button>
                  </span>
                  <div
                    class="euiText euiText--small euiCard__description"
                    id="random_html_idDescription"
                  >
                    <p />
                  </div>
                  <div
                    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
                  >
                    <div
                      class="euiFlexItem"
                    >
                      <dl
                        class="euiDescriptionList euiDescriptionList--row"
                      >
                        <dt
                          class="euiDescriptionList__title"
                          style="color: rgb(105, 112, 125); font-size: 0.6875rem; font-weight: normal;"
                        >
                          Channel type
                        </dt>
                        <dd
                          class="euiDescriptionList__description"
                        >
                          email
                        </dd>
                      </dl>
                    </div>
                    <div
                      class="euiFlexItem"
                    >
                      <dl
                        class="euiDescriptionList euiDescriptionList--row"
                      >
                        <dt
                          class="euiDescriptionList__title"
                          style="color: rgb(105, 112, 125); font-size: 0.6875rem; font-weight: normal;"
                        >
                          Sent status
                        </dt>
                        <dd
                          class="euiDescriptionList__description"
                        >
                          <div
                            class="euiHealth"
                          >
                            <div
                              class="euiFlexGroup euiFlexGroup--gutterExtraSmall euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow"
                            >
                              <div
                                class="euiFlexItem euiFlexItem--flexGrowZero"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="euiIcon euiIcon--medium euiIcon--danger euiIcon-isLoading"
                                  focusable="false"
                                  height="16"
                                  role="img"
                                  viewBox="0 0 16 16"
                                  width="16"
                                  xmlns="http://www.w3.org/2000/svg"
                                />
                              </div>
                              <div
                                class="euiFlexItem euiFlexItem--flexGrowZero"
                              >
                                <div
                                  class="euiText euiText--small"
                                >
                                  Error
                                </div>
                              </div>
                            </div>
                          </div>
                        </dd>
                      </dl>
                    </div>
                  </div>
                  <dl
                    class="euiDescriptionList euiDescriptionList--row"
                  >
                    <dt
                      class="euiDescriptionList__title"
                      style="color: rgb(105, 112, 125); font-size: 0.6875rem; font-weight: normal;"
                    >
                      Error details
                    </dt>
                    <dd
                      class="euiDescriptionList__description"
                    >
                      failed
                    </dd>
                  </dl>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlyoutFooter"
      >
        <button
          class="euiButtonEmpty euiButtonEmpty--primary euiButtonEmpty--flushLeft"
          type="button"
        >
          <span
            class="euiButtonContent euiButtonEmpty__content"
          >
            <svg
              aria-hidden="true"
              class="euiIcon euiIcon--medium euiIcon-isLoading euiButtonContent__icon"
              focusable="false"
              height="16"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            />
            <span
              class="euiButtonEmpty__text"
            >
              Close
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
    data-focus-guard="true"
    style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
    tabindex="0"
  />
</div>
`;
